src/Block/MediaBlockService.php

```diff
@@ -209,10 +204,10 @@ protected function getFormatChoices(?MediaInterface $media = null): array
     protected function getMediaBuilder(FormMapper $form): FormBuilderInterface
     {
         // simulate an association ...
-        $fieldDescription = $this->getMediaAdmin()->getModelManager()->getNewFieldDescriptionInstance($this->mediaAdmin->getClass(), 'media', [
+        $fieldDescription = $this->fieldDescriptionFactory->create($this->mediaAdmin->getClass(), 'media', [
             'translation_domain' => 'SonataMediaBundle',
         ]);
-        $fieldDescription->setAssociationAdmin($this->getMediaAdmin());
+        $fieldDescription->setAssociationAdmin($this->mediaAdmin);
         $fieldDescription->setAdmin($form->getAdmin());
         $fieldDescription->setOption('edit', 'list');
         $fieldDescription->setAssociationMapping([
```

Comment:
This method was removed. This is now in the construct of the fieldDescription.
And for instance the DoctrineORMAdmin fieldDescriptionFactory->create method is doing
```
[$metadata, $propertyName, $parentAssociationMappings] = $this->getParentMetadataForProperty($class, $name);

        return new FieldDescription(
            $name,
            $options,
            $metadata->fieldMappings[$propertyName] ?? [],
            $metadata->associationMappings[$propertyName] ?? [],
            $parentAssociationMappings,
            $propertyName
        );
```

So I would say this call can be removed, and it's the fieldDescriptionFactory to correctly set the association mapping.

Reply:
Removed